File: python/day01.py
import fileinput
data = list()
for line in fileinput.input():
    data.append(int(line))

n = 0
for x in data:
    n += x
part1 = n

part2 = None
H = set()
n = 0
while part2 is None:
    for x in data:
        n += x
        if H.__contains__(n):
            part2 = n
            break
        H.add(n)

print(part1)
print(part2)

File: src/main/java/aoc/y2021/Day01.java
package aoc.y2021;

import static java.lang.System.in;
import static java.util.Arrays.stream;

public class Day01 {
    public static void main(String...args) throws Exception {
        var data = stream(new String(in.readAllBytes()).split("\n"))
                .mapToInt(Integer::parseInt)
                .toArray();

        var part1 = 0;
        for (int i = 1; i < data.length; ++i) {
            if (data[i] > data[i-1]) {
                part1++;
            }
        }

        System.out.println(part1);

        var part2 = 0;
        for (int i = 3; i < data.length; ++i) {
            if (data[i] > data[i - 3]) {
                part2++;
            }
        }

        System.out.println(part2);
    }
}
